# 25 · GSE135779 · scRNA_seq · WGCNA modules

Reads `wgcna_input.rds`. Writes `modules.rds` and the end product `module_genes.csv` in
`data/run_artifacts/GSE135779/`.

Settings as for GSE65391 (notebook 05): signed network and TOM, minModuleSize 30, mergeCutHeight 0.25.
There are more genes than fit one block, so `maxBlockSize = 5000`: WGCNA pre-clusters the genes into
blocks and builds modules within each block.

## Load WGCNA and the input from notebook 24

In [1]:
if (!requireNamespace("WGCNA", quietly = TRUE)) install.packages("WGCNA", repos = "https://cloud.r-project.org")
suppressMessages(library(WGCNA))
source("../src/paths.R")
w <- readRDS(art("GSE135779", "wgcna_input.rds"))
c(samples = nrow(w$datExpr), genes = ncol(w$datExpr), power = w$power)

samples   genes   power 
     33   15353      18

**Explanation**
- As in notebook 05, with the input of notebook 24.

## Find the modules

In [2]:
set.seed(SEED)
net <- blockwiseModules(w$datExpr, power = w$power, networkType = "signed", TOMType = "signed",
                        minModuleSize = 30, mergeCutHeight = 0.25, maxBlockSize = 5000,
                        numericLabels = FALSE, pamRespectsDendro = FALSE, verbose = 0)
sizes <- sort(table(net$colors), decreasing = TRUE)
c(modules = sum(names(sizes) != "grey"), genes_in_modules = sum(sizes[names(sizes) != "grey"]), grey = unname(sizes["grey"]))
sizes

modules genes_in_modules             grey 
              30            13943             1410


    turquoise          blue          grey         brown        yellow 
         2580          2340          1410          1098           902 
        green           red         black          pink       magenta 
          819           722           661           606           527 
       purple   greenyellow           tan        salmon          cyan 
          455           333           330           328           310 
 midnightblue     lightcyan        grey60    lightgreen   lightyellow 
          248           225           218           193           180 
    royalblue       darkred     darkgreen darkturquoise      darkgrey 
          171           124           120            99            65 
       orange    darkorange         white       skyblue   saddlebrown 
           65            55            51            47            37 
    steelblue 
           34 

**Explanation**
- As in notebook 14, including `maxBlockSize = 5000`.

**Result.** 30 modules holding 13,943 of 15,353 genes; 1,410 are grey.

## Module membership and hub genes

In [3]:
ME  <- orderMEs(net$MEs)
kME <- cor(w$datExpr, ME)
modules <- setdiff(names(sizes), "grey")
hubs <- lapply(setNames(modules, modules), function(mo) {
  g <- names(net$colors)[net$colors == mo]
  names(sort(kME[g, paste0("ME", mo)], decreasing = TRUE))
})
t(sapply(hubs, function(h) paste(head(h, 8), collapse = ", ")))

turquoise,blue,brown,yellow,green,red,black,pink,magenta,purple,⋯,darkred,darkgreen,darkturquoise,darkgrey,orange,darkorange,white,skyblue,saddlebrown,steelblue
"PNISR, DPH5, ATM, BCL11B, RPSA, PRKCQ-AS1, POLR1E, RPL31","EMILIN2, RXRA, NOTCH2, PICALM, TTYH3, ATP11A, DMXL2, UBTD1","BOLA1, LEO1, METTL13, TMEM223, SNUPN, LRIF1, MIS18A, SPDL1","PRXL2B, CDC25B, TAF6, TMEM129, ARHGAP45, ESYT1, TBC1D10C, PRAF2","FBXO33, CHD1, RLF, JOSD1, CCNL1, KBTBD2, ZNF394, FAM133B","CCNB1, RAD51, ASF1B, CDCA8, SPC25, CDK1, CDCA3, CCNA2","TMEM259, SIPA1, SIAH2, PTOV1, HDAC10, YRDC, ZNF865, DCTN1","HNRNPA0, PHF1, VAMP2, TLE5, ABHD14B, MOAP1, SEPTIN9, CCNG1","IFI35, HESX1, FBXO6, LAP3, STAT2, SERPING1, ISG15, CHMP5","ATP5PD, COX7A2, COXFA4, SUMO2, CHCHD2, MRPS15, PSMB7, NEDD8",⋯,"ZNF14, BCL7B, RAB11FIP4, PIP4P1, NR1H2, ARRDC2, DUSP11, CLK3","CMTM5, TUBB1, SPARC, ACRBP, CTTN, CAVIN2, GP9, GFI1B","PSMB3, POMP, ATOX1, SUPT4H1, CHMP2A, BLOC1S1, VAMP5, LAMTOR2","SETX, TMEM131, EFR3A, LCORL, MAPK14, WASHC4, QSER1, TMED8","CDC5L, ASCC3, TMOD3, XRN1, BAZ1A, ARCN1, DHX29, CEP295","LAMTOR4, ATP5MJ, POLE4, GPX4, ELOB, ATP5PB, SEC11A, ATP5MC2","TMIGD3, SRGAP1, SAP30, ZBTB16, SLC1A3, ALOX15B, NUDT16, CD163","CA1, SLC4A1, SNCA, SELENBP1, LINC00570, TMC5-AS1, ALAS2, GYPB","LMNB1, CGAS, MTHFD2, SPPL2A, EZH2, MTFR2, MICB, FAS","HSPA13, TUBB4B, KPNA2, TMBIM6, TFG, HSPA5, SURF4, GDE1"


**Explanation**
- As in notebook 05.

## Which modules hold the interferon genes?

In [4]:
ifn6 <- read_gene_set("ifn-type1-6.txt")
data.frame(gene = ifn6, module = net$colors[ifn6], row.names = NULL)

gene,module
<chr>,<chr>
IFI27,magenta
IFI44L,magenta
IFIT1,magenta
ISG15,magenta
RSAD2,magenta
SIGLEC1,magenta


**Explanation**
- As in notebook 05.

**Result.** All six interferon-score genes are in one module, magenta (hub genes IFI35, HESX1, FBXO6, LAP3,
STAT2). Other modules read by their hub genes: red (CCNB1, RAD51, CDK1: cell cycle), salmon (FCRLA, MS4A1,
CD19, CD79B: B cell), grey60 (CCL5, CST7, FCRL6: NK / cytotoxic), darkgreen (CMTM5, TUBB1, SPARC:
platelet), skyblue (CA1, SLC4A1, SELENBP1: red cell) and turquoise (BCL11B, RPSA: lymphoid and ribosomal).

## Save the modules

In [5]:
in_module <- names(net$colors)[net$colors != "grey"]
module_genes <- data.frame(gene = in_module, module = net$colors[in_module],
                           kME = kME[cbind(in_module, paste0("ME", net$colors[in_module]))], row.names = NULL)
write.csv(module_genes, art("GSE135779", "module_genes.csv"), row.names = FALSE)
saveRDS(list(colors = net$colors, MEs = ME, kME = kME, hubs = hubs), art("GSE135779", "modules.rds"))

**Explanation**
- As in notebook 05. These are this study's own modules, not the array modules.